### ISSUE TITLE
Initializing object layer failed with XL format

### ISSUE THREAD

**Issue by jblunck** [minio/minio#5645]
I'm trying to run Minio bare-metal on 7 nodes with 4 disks each. Although the metabuckets seem to get created correctly (according to my understanding of the code) starting the server fails with the following error:

```
root@dell03-svr02:/opt/minio# ./minio server --config-dir /etc/minio --address 10.1.40.61:9000 http://dell03-svr0{2...8}.pod/opt/minio/data_nvme{0...3}n1


Trace: 1: /q/.q/sources/gopath/src/github.com/minio/minio/cmd/server-main.go:246:cmd.serverMain()
       2: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:499:cli.HandleAction()
       3: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/command.go:214:cli.Command.Run()
       4: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:260:cli.(*App).Run()
       5: /q/.q/sources/gopath/src/github.com/minio/minio/cmd/main.go:155:cmd.Main()
       6: /q/.q/sources/gopath/src/github.com/minio/minio/main.go:71:main.main()
[2018-03-13T13:25:44.874647637Z] [ERROR] Initializing object layer failed (Unable to initialize '.minio.sys' meta volume, disk access denied)
```

I've also tried other command lines that result in the same error:
```
root@dell03-svr02:/opt/minio# rm -vfr data_nvme*/.minio.sys
removed directory 'data_nvme0n1/.minio.sys/tmp'
removed directory 'data_nvme0n1/.minio.sys/multipart'
removed directory 'data_nvme0n1/.minio.sys'
removed directory 'data_nvme1n1/.minio.sys/tmp'
removed directory 'data_nvme1n1/.minio.sys/multipart'
removed directory 'data_nvme1n1/.minio.sys'
removed directory 'data_nvme2n1/.minio.sys/tmp'
removed directory 'data_nvme2n1/.minio.sys/multipart'
removed directory 'data_nvme2n1/.minio.sys'
removed directory 'data_nvme3n1/.minio.sys/tmp'
removed directory 'data_nvme3n1/.minio.sys/multipart'
removed directory 'data_nvme3n1/.minio.sys'
root@dell03-svr02:/opt/minio# ./minio server --config-dir /etc/minio --address 10.1.40.61:9000 http://dell03-svr02.pod/opt/minio/data_nvme0n1 http://dell03-svr02.pod/opt/minio/data_nvme1n1 http://dell03-svr02.pod/opt/minio/data_nvme2n1 http://dell03-svr02.pod/opt/minio/data_nvme3n1 http://dell03-svr03.pod/opt/minio/data_nvme0n1 http://dell03-svr03.pod/opt/minio/data_nvme1n1 http://dell03-svr03.pod/opt/minio/data_nvme2n1 http://dell03-svr03.pod/opt/minio/data_nvme3n1 http://dell03-svr04.pod/opt/minio/data_nvme0n1 http://dell03-svr04.pod/opt/minio/data_nvme1n1 http://dell03-svr04.pod/opt/minio/data_nvme2n1 http://dell03-svr04.pod/opt/minio/data_nvme3n1 http://dell03-svr05.pod/opt/minio/data_nvme0n1 http://dell03-svr05.pod/opt/minio/data_nvme1n1 http://dell03-svr05.pod/opt/minio/data_nvme2n1 http://dell03-svr05.pod/opt/minio/data_nvme3n1 http://dell03-svr06.pod/opt/minio/data_nvme0n1 http://dell03-svr06.pod/opt/minio/data_nvme1n1 http://dell03-svr06.pod/opt/minio/data_nvme2n1 http://dell03-svr06.pod/opt/minio/data_nvme3n1 http://dell03-svr07.pod/opt/minio/data_nvme0n1 http://dell03-svr07.pod/opt/minio/data_nvme1n1 http://dell03-svr07.pod/opt/minio/data_nvme2n1 http://dell03-svr07.pod/opt/minio/data_nvme3n1 http://dell03-svr08.pod/opt/minio/data_nvme0n1 http://dell03-svr08.pod/opt/minio/data_nvme1n1 http://dell03-svr08.pod/opt/minio/data_nvme2n1 http://dell03-svr08.pod/opt/minio/data_nvme3n1


Trace: 1: /q/.q/sources/gopath/src/github.com/minio/minio/cmd/server-main.go:246:cmd.serverMain()
       2: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:499:cli.HandleAction()
       3: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/command.go:214:cli.Command.Run()
       4: /q/.q/sources/gopath/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:260:cli.(*App).Run()
       5: /q/.q/sources/gopath/src/github.com/minio/minio/cmd/main.go:155:cmd.Main()
       6: /q/.q/sources/gopath/src/github.com/minio/minio/main.go:71:main.main()
[2018-03-13T13:34:22.730680676Z] [ERROR] Initializing object layer failed (Unable to initialize '.minio.sys' meta volume, disk access denied)
```

## Your Environment

* Version used (`minio version`):
Version: 2018-03-12T21:25:28Z
Release-Tag: RELEASE.2018-03-12T21-25-28Z
Commit-ID: 4a4d1d1b82046a2af62381434bcbff1d748db477
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version: Dell PowerEdge R740xd
* Operating System and version (`uname -a`): Linux dell03-svr02 4.13.0-36-generic #40-Ubuntu SMP Fri Feb 16 20:07:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
* Link to your project:

**Comment by harshavardhana** [comment]
Thanks, @jblunck for the call. To summarize the issues we found were 

- Permission issues on the backend disks v/s the user running the minio server
- `--config-dir` option before and after the `server` command line can cause discrepancies 

We will try to reproduce the problem on our end as well related to `--config-dir`

**Comment by harshavardhana** [comment]
This doesn't seem like a bug @jblunck `--config-dir` works in either position. The problem is mostly due to simply permissions and the "signature invalid" was because on port 9000 other remote servers were trying to communicate with different access keys.

Going to send another PR where we will return a higher level error when disks are not writable.

**Comment by nitisht** [comment]
Fixed by #5662
